### ISSUE TITLE
setuid and setgid take locks on Linux

### ISSUE THREAD

**Issue by DemiMarie** [haskell/process#83]
On Linux, setuid and setgid both take locks, so neither is safe to call in the child after vfork().  Instead, use syscall to issue a raw system call to the kernel.

**Comment by snoyberg** [comment]
I at least don't have enough context to understand this bug report, can you clarify?

**Comment by simonmichael** [comment]
I wonder if this is related to the fact that rpmlint warns about missing-call-to-setgroups-before-setuid with some haskell executables, such as hledger-web (https://github.com/simonmichael/hledger/issues/1057) and ghcid (https://bugzilla.redhat.com/show_bug.cgi?id=1639089) ?

**Comment by bgamari** [comment]
I don't believe this is the same issue that rpmlint is complaining about. That being said, I do believe that the rpmlint warning is a real issue that may even lead to privilege escalation. I've fixed this in #148.
